### PR TITLE
feat: add error handling and loading states

### LINKED ISSUE #141: [improve]: add error handling and loading states
Currently there is no error handling logic in the fetch requests and no loading states in the application. 

TODO: 
- Add catch clauses in fetch
- Add loading state to render a progressbar

### PR DESCRIPTION
**Pull request by toth2000**
Fixes #141 

## Proposed Changes

- Added
    - Loading progress bar on Home component
    - Catch clauses in fetch 

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/155"></a>

### CODE REVIEW

File: src/Components/Home.js [hunk 1]
```diff
@@ -34,6 +40,7 @@ function Home() {
           onClick={() => goToLinkHandle(user.username)}
         />
       ))}
+      {showProgress && <progress />}
```
Comment (Cahllagerfeld) [inline src/Components/Home.js:43]: We are using PrimeReact, which Provides a progressbar out of the box. We should use this, to be consistent in the UI.
See the docs for the progressbar [here](https://primefaces.org/primereact/showcase/#/progressbar)

**Review by Cahllagerfeld — CHANGES_REQUESTED** [review]
Thanks for your contribution 👍 

I left one small comment regarding the component you used.

File: src/Components/Home.js [hunk 2]
```diff
@@ -21,6 +22,11 @@ function Home() {
     fetch('/data/_list.json')
       .then((response) => response.json())
       .then((data) => setList(data))
+      .catch((error) => {
+        console.log('Home useEffect', error)
+        alert('An error occurred please try again later')
```
Comment (schmelto) [inline src/Components/Home.js:27]:
```suggestion
        alert('An error occurred please try again later.')
```

File: src/Components/Home.js (hunk 1 above)
Comment (toth2000) [reply src/Components/Home.js]: Thanks, @Cahllagerfeld for the feedback. I have made the suggested changes.

**Review by BOLT04 — APPROVED** [review]
LGTM 👍

File: src/Components/Home.js [hunk 3]
```diff
@@ -32,6 +39,7 @@ function Home() {
           />
         </a>
       ))}
+      {showProgress && <ProgressBar mode="indeterminate" />}
```
Comment (eddiejaoude) [inline src/Components/Home.js:42]: Should this also go on the Socials component when loading a specific user? I will merge, but another PR could be done to be consistent

**Pull request merged**